File: Manual 01/Question06.cpp
#include <iostream>
using namespace std;

int main()
{
    int x;
    cout << "Input a number: ";
    cin >> x;
    cout << endl;

    // Check if the number is non-negative
    if (x >= 0)
    {
        // Determine the range and whether the number is odd or even
        if (x <= 50)
        {
            if (x % 2 == 1)
            {
                cout << "The input is odd" << endl;
            }
            else
            {
                cout << "The input is even" << endl;
            }
        }
        else
        {
            // For numbers greater than 50
            if (x % 2 == 1)
            {
                cout << "The input is odd" << endl;
            }
            else
            {
                cout << "The input is even" << endl;
            }
        }
    }
    else
    {
        cout << "Invalid input. Please enter a non-negative number." << endl;
    }

    return 0;
}

File: Python/Manual01.py/Question06.py
number = int(input("Enter a Number: "))

if number < 0:
    print("Enter a Positive Number")
else:
    if number <= 10:
        print("Number is between 0 and 10")
    elif number <= 20:
        print("Number is between 11 and 20")
    elif number <= 30:
        print("Number is between 21 and 30")
    elif number <= 40:
        print("Number is between 31 and 40")
    elif number <= 50:
        print("Number is between 41 and 50")
    else:
        print("Number is greater than 50")

    if number % 2 == 0:
        print("Number is Even")
    else:
        print("Number is Odd")
